src/cim/components/InfraEnv/InfraEnvHostsTabAgentsWarning.tsx

```diff
@@ -1,28 +1,54 @@
 import * as React from 'react';
 import { ExclamationTriangleIcon } from '@patternfly/react-icons';
 import { global_warning_color_100 as warningColor } from '@patternfly/react-tokens';
-import { getFailingAgentConditions } from '../helpers';
-import { AgentK8sResource } from '../../types';
+import { getAgentStatus, getBMHStatus } from '../helpers';
+import { AgentK8sResource, BareMetalHostK8sResource } from '../../types';
+import { AGENT_BMH_NAME_LABEL_KEY } from '../common';
 
 export type InfraEnvHostsTabAgentsWarning = {
   infraAgents: AgentK8sResource[];
+  infraBMHs: BareMetalHostK8sResource[];
 };
 
 const InfraEnvHostsTabAgentsWarning: React.FC<InfraEnvHostsTabAgentsWarning> = ({
   infraAgents,
+  infraBMHs,
 }) => {
-  const faiingAgentConditions = getFailingAgentConditions(infraAgents);
-  const failingAgentNames = Object.getOwnPropertyNames(faiingAgentConditions);
-  if (!failingAgentNames.length) {
-    return null;
-  }
+  const agentsBMHNames = infraAgents.reduce<string[]>((names, agent) => {
+    const name = agent.metadata?.labels?.[AGENT_BMH_NAME_LABEL_KEY];
+    name && names.push(name);
+    return names;
+  }, []);
 
-  return (
-    <ExclamationTriangleIcon
-      color={warningColor.value}
-      className="infra-env-hosts-tab-title__icon"
-    />
+  const bmhsWithoutAgents = infraBMHs.filter((bmh) =>
+    agentsBMHNames.includes(bmh.metadata?.name || ''),
   );
+
+  const agentStates = infraAgents.map((agent) => getAgentStatus(agent, true).status);
+  const bmhStates = bmhsWithoutAgents.map((bmh) => getBMHStatus(bmh).state);
+
+  const hostStates = [...agentStates, ...bmhStates];
+
+  const problemStates = [
```

Comment:
how about adding `type` to `HostStatusDef` ? So we have all these in one place

Reply:
Quite often the list differs based on the specific use case. In other words, there will be cases when the list of states belonging to a certain status type will differ and the type property on the HostStatusDef would not be correct. So I am inclined to keeping the list just here.